docs/advanced-features/compiler.md

```diff
@@ -331,30 +361,77 @@ module.exports = {
 }
 ```
 
-Cause this code:
+The above config will transform your code as follows:
 
 ```js
+// Before
 import { MyModule } from 'my-library'
 import { App } from 'my-library/components'
 import { Header, Footer } from 'my-library/components/App'
-```
-
-To become:
 
-```js
+// After
 import MyModule from 'my-library/MyModule'
 import App from 'my-library/components/App'
 import Header from 'my-library/components/App/Header'
 import Footer from 'my-library/components/App/Footer'
 ```
 
-- Handlebars templating
+#### Using named imports
 
-This transform uses [handlebars](https://docs.rs/handlebars) to template the replacement import path in the `transform` field. These variables and helper functions are available:
+By default, `modularizeImports` assumes that each module uses default exports. However, this may not always be the case — named exports may be used.
+
+```js
+// my-library/MyModule.ts
+// Using named export instead of default export
+export const MyModule = {}
+
+// my-library/index.ts
+// The “barrel file” that re-exports `MyModule`
+export { MyModule } from './MyModule'
+```
 
-1. `matches`: Has type `string[]`. All groups matched by the regular expression. `matches.[0]` is the full match.
-2. `member`: Has type `string`. The name of the member import.
-3. `lowerCase`, `upperCase`, `camelCase`, `kebabCase`: Helper functions to convert a string to lower, upper, camel or kebab cases.
+In this case, you can use the `skipDefaultConversion` option to use named imports instead of default imports:
+
+```js
+// next.config.js
+module.exports = {
+  'my-library': {
+    transform: 'my-library/{{member}}',
+    skipDefaultConversion: true,
+  },
+}
+```
+
+The above config will transform your code as follows:
+
+```js
+// Before
+import { MyModule } from 'my-library'
+
+// After (imports `MyModule` using named import)
+import { MyModule } from 'my-library/MyModule'
+```
+
+#### Preventing full import
```

Comment:
Added docs for `preventFullImport` ([Next.js type](https://github.com/vercel/next.js/blob/90b86be6dc50287b589363dc71a89f8bb7600736/packages/next/src/server/config-shared.ts#L506) / [SWC implementation](https://github.com/swc-project/plugins/blob/221ec0d713346c5460cae6c789f29ff9dcb07897/packages/transform-imports/transform/src/lib.rs#L101)) which was not documented.

I tested this locally.